dbt/adapters/duckdb/relation.py

```diff
@@ -34,6 +37,21 @@ def create_from_source(cls: Type[Self], source: ParsedSourceDefinition, **kwargs
             ext_location = ext_location.format(
                 schema=source.schema, name=source.name, identifier=source.identifier
             )
+            if ".xlsx" in ext_location:
+
+                # Strip quotes.
+                if ext_location.startswith("'"):
+                    ext_location = ext_location[1:-1]
+
+                df = pd.read_excel(ext_location)
+                TEMP_DIR = pathlib.Path("/tmp/dbt-excel/")
```

Comment:
I was thinking that we could create a directory at the same location and with the same name as the Excel file